tests/unit_and_functional/test_workflow_execution.py

```diff
@@ -161,3 +169,56 @@ def test_run_workflow_adds_git_commit_to_batch_and_service_env_vars(
         cpu_request=ANY,
         memory_request=ANY,
     )
+
+
+@patch("bodywork.workflow_execution.run")
+@patch("bodywork.workflow_execution.rmtree")
+@patch("bodywork.workflow_execution.requests")
+@patch("bodywork.workflow_execution.download_project_code_from_repo")
+@patch("bodywork.workflow_execution.get_git_commit_hash")
+@patch("bodywork.workflow_execution.k8s")
+def test_run_workflow_pings_usage_stats_server(
+    mock_k8s: MagicMock,
+    mock_git_hash: MagicMock,
+    mock_git_download: MagicMock,
+    mock_requests: MagicMock,
+    mock_rmtree: MagicMock,
+    mock_run: MagicMock,
+    project_repo_location: Path,
+):
+    param = "-n" if platform.system() == "Windows" else "-c"
+    config_path = Path(f"{project_repo_location}/bodywork.yaml")
+    config = BodyworkConfig(config_path)
+
+    run_workflow(config, "foo_bar_foo_993", project_repo_location)
+
+    mock_run.assert_called_once_with(
+        ["ping", param, "1", USAGE_STATS_SERVER_URL],
+        check=True,
+        capture_output=True,
+        encoding="utf-8",
+    )
+
+
+@patch("bodywork.workflow_execution.run")
+@patch("bodywork.workflow_execution.rmtree")
+@patch("bodywork.workflow_execution.requests")
+@patch("bodywork.workflow_execution.download_project_code_from_repo")
+@patch("bodywork.workflow_execution.get_git_commit_hash")
+@patch("bodywork.workflow_execution.k8s")
+def test_usage_stats_opt_out_does_not_ping_usage_stats_server(
+    mock_k8s: MagicMock,
+    mock_git_hash: MagicMock,
+    mock_git_download: MagicMock,
+    mock_requests: MagicMock,
+    mock_rmtree: MagicMock,
+    mock_run: MagicMock,
+    project_repo_location: Path,
+):
+    config_path = Path(f"{project_repo_location}/bodywork.yaml")
+    config = BodyworkConfig(config_path)
+    config.project.usage_stats = False
+
+    run_workflow(config, "foo_bar_foo_993", project_repo_location)
+
+    mock_run.assert_not_called()
```

Comment:
Need a newline at the end of the file - flake8 should be complaining about that 😬

Is flake8 working post-CICD pipeline refactor?

Reply:
Good spot! I have no idea why Flake8 didn't pick that up as it definitely seems to be running. It has got me on other flake8 errors recently.